homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
+SELECT *
+FROM employees
+WHERE salary BETWEEN (
+    SELECT MIN(salary)
+    FROM employees
+    ) AND 2500;
+
 --21. all the information of the employees who does not work in those departments where some employees works whose id within the range 100 and 200.
+SELECT *
+      FROM employees
+      WHERE department_id NOT IN (
+          SELECT department_id
+          FROM employees
+                   INNER JOIN departments USING (department_id)
+          WHERE employee_id BETWEEN 100 AND 200
+      );
+
 --22. all the information for those employees whose id is any id who earn the second highest salary.
+SELECT *
+FROM employees
+WHERE salary IN (SELECT MAX(salary)
+                 FROM employees
+                 WHERE salary != (SELECT MAX(salary)
+                                  FROM employees)
+);
+
 --23. the employee name( first name and last name ) and hiredate for all employees in the same department as Clara. Exclude Clara.
+SELECT (first_name || ' ' || last_name) AS employee_name, hire_date
+FROM employees
+WHERE department_id = (
+    SELECT department_id
+    FROM employees
+    WHERE first_name = 'Clara'
+    ) AND first_name != 'Clara';
+
 --24. the employee number and name( first name and last name ) for all employees who work in a department with any employee whose name contains a T.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name
+FROM employees
+WHERE department_id in (
+    SELECT department_id
+    FROM employees
+    WHERE (first_name || ' ' || last_name) LIKE '%T%'
+);
+
 --25. full name(first and last name), job title, starting and ending date of last jobs for those employees with worked without a commission percentage.
+SELECT (first_name || ' ' || last_name) AS employee_name, job_title, last_start_date, last_end_date
+FROM employees
+    INNER JOIN (
+        SELECT employee_id, MAX (start_date) as last_start_date, MAX (end_date) as last_end_date
+        FROM job_history
+        GROUP BY employee_id
+    ) USING (employee_id)
+    INNER JOIN jobs USING (job_id)
+WHERE commission_pct IS NULL;
+
 --26. the employee number, name( first name and last name ), and salary for all employees who earn more than the average salary and who work in a department with any employee with a J in their name.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name, salary
+FROM employees
+WHERE salary > (
+    SELECT AVG(salary)
+    FROM employees
+)
+  AND department_id in (
+    SELECT department_id
+    FROM employees
+    WHERE (first_name || ' ' || last_name) LIKE '%J%'
+);
+
 --27. the employee number, name( first name and last name ) and job title for all employees whose salary is smaller than any salary of those employees whose job title is MK_MAN.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name, job_title
+FROM employees
+    INNER JOIN jobs USING (job_id)
+WHERE salary < (
+    SELECT MIN(salary)
+    FROM employees
+    WHERE job_id = 'MK_MAN'
+);
```

Comment:
I see in condition "is smaller than **any**". I guess you can add it too. 
```suggestion
WHERE salary < ANY (
    SELECT MIN(salary)
    FROM employees
    WHERE job_id = 'MK_MAN'
);
```

Reply:
Then, there is no need no use MIN on **salary**.
Refactored to use **ANY**